test/e2e/basic_test.go

```diff
@@ -110,15 +116,27 @@ func TestPodLifecycle(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// Wait for the "nginx-1-Y" pod to be deleted in a separate goroutine.
+	// This ensures that we don't possibly miss the MODIFIED/DELETED events due to establishing the watch too late in the process.
+	pod1Ch := make(chan struct{})
+	go func() {
+		// Wait for the "nginx-1-Y" pod to be reported as having been marked for deletion.
+		if err := f.WaitUntilPodDeleted(pod1.Namespace, pod1.Name); err != nil {
+			t.Fatal(err)
+		} else {
+			// Close the pod0Ch channel, signaling we've observed deletion of the pod.
+			close(pod1Ch)
+		}
+	}()
+
 	// Delete the "nginx-1" pod.
 	if err := f.DeletePod(pod1.Namespace, pod1.Name); err != nil {
 		t.Fatal(err)
 	}
-
-	// Wait for the "nginx-1-Y" pod to be reported as having been marked for deletion.
-	if err := f.WaitUntilPodDeleted(pod1.Namespace, pod1.Name); err != nil {
-		t.Fatal(err)
-	}
+	// Wait for the delete event to be ACKed.
+	<-pod1Ch
```

Comment:
I have amended the test. If `WaitUntilPodDeleted` returns an error - and it will if no event is observed after 2m - the test now fails.

Done in 3a0874c.